access/ms-teams/bot.go

```diff
@@ -125,52 +136,72 @@ func (b Bot) UninstallAppForUser(ctx context.Context, userIDOrEmail string) erro
 	return trace.Wrap(err)
 }
 
-// FetchUser fetches app id for user, installs app for a user if missing, fetches chat id and saves
-// everything to cache. This method is used for priming the cache. Returns trace.NotFound if a
+// FetchRecipient checks if recipient is a user or a channel, installs app for a user if missing, fetches chat id
+// and saves everything to cache. This method is used for priming the cache. Returns trace.NotFound if a
 // user was not found.
-func (b Bot) FetchUser(ctx context.Context, userIDOrEmail string) (*UserData, error) {
+func (b Bot) FetchRecipient(ctx context.Context, recipient string) (*RecipientData, error) {
 	if b.teamsApp == nil {
 		return nil, trace.Errorf("Bot is not configured, run GetTeamsApp first")
 	}
 
 	b.mu.RLock()
-	d, ok := b.users[userIDOrEmail]
+	d, ok := b.recipients[recipient]
 	b.mu.RUnlock()
 	if ok {
 		return &d, nil
 	}
 
-	userID, err := b.getUserID(ctx, userIDOrEmail)
-	if err != nil {
-		return &UserData{}, trace.Wrap(err)
-	}
-
-	var installedApp *msapi.InstalledApp
-
-	installedApp, err = b.graphClient.GetAppForUser(ctx, b.teamsApp, userID)
-	if trace.IsNotFound(err) {
-		err := b.graphClient.InstallAppForUser(ctx, userID, b.teamsApp.ID)
+	// Check if the recipient is a channel
+	channel, isChannel := b.checkChannelURL(recipient)
+	if isChannel {
+		// A team and a group are different but in MsTeams the team is associated to a group and will have the same id.
+		installedApp, err := b.graphClient.GetAppForTeam(ctx, b.teamsApp, channel.Group)
```

Comment:
When we do the same for the user we also create the app_user chat
The same doesn't happen here: we only "get" the channel but there's no creation if it doesn't exist

Reply:
So, it might happen that we try to use a channel but it won't work because the Admin didn't grant permission for it when installing the bot :thinking: 

If so, I think we should add a check for write access for that particular channel when starting up the app (when `preload=true`)

Reply:
But, FetchRecipient is already called during preload, and the check logic is in FetchRecipient. Maybe I'm entirely missing the point but I think we already are doing this preload check

Reply:
Oh right :+1: 
Thank you :bow: